apps/personal/contacts/main.py

```diff
@@ -4,57 +4,113 @@
         self.menu_name = "Contacts"
         self.address_book = AddressBook()
         self.menu = None
+        self.vdirsyncer_executable = find_executable('vdirsyncer')
 
     def on_start(self):
+        self.reload()
+
+    def reload(self):
         self.address_book.load_from_file()
-        self.menu = NumberedMenu(self.create_menu_content(), self.i, self.o, prepend_numbers=False)
+        self.menu = NumberedMenu(self.build_main_menu_content(), i=self.i,
+                                 o=self.o, prepend_numbers=False)
         self.menu.activate()
 
-    def create_menu_content(self):
+    def build_main_menu_content(self):
         all_contacts = self.address_book.contacts
-        return [[c.short_name(), lambda x=c: self.create_contact_page(x)] for c in all_contacts]
+        menu_entries = []
+        menu_entries.append(["|| Actions", lambda: self.open_actions_menu()])
+        for c in all_contacts:
+            menu_entries.append([c.short_name(), lambda x=c:
+                                 self.open_contact_details_page(x)])
+
+        return menu_entries
 
-    def create_contact_page(self, contact):
+    def open_contact_details_page(self, contact):
         # type: (Contact) -> None
-        contact_attrs = [getattr(contact, a) for a in contact.get_filled_attributes()]
+        contact_attrs = [getattr(contact, a) for a in
+                         contact.get_filled_attributes()]
         Listbox(i=self.i, o=self.o, contents=contact_attrs).activate()
 
-
-def find_contact_files(folder):
-    # type: (str) -> list(str)
-    home = os.path.expanduser(folder)
-    if not os.path.exists(home):
-        os.mkdir(home)
-    contact_card_files = [os.path.join(home, f) for f in os.listdir(home) if f.lower().endswith("vcf")]
-    return contact_card_files
-
-
-def load_vcf(folder):
-    # type: (str) -> None
-    contact_card_files = find_contact_files(folder)
-    contacts = VCardContactConverter.from_vcards(contact_card_files)
-
-    address_book = AddressBook()
-    for contact in contacts:
-        address_book.add_contact(contact)
-    address_book.save_to_file()
-    logger.info("Saved to {}".format(address_book.get_save_file_path()))
-
+    def open_actions_menu(self):
+        menu_contents = [
+            ["Settings", lambda: self.open_settings_page()],
+            ["CardDAV Import", lambda: self.synchronize_carddav()],
+            ["Reset address book", lambda: self.reset_addressbook()]
+        ]
+        Menu(menu_contents, i=self.i, o=self.o, name="My menu").activate()
+
+    def open_settings_page(self):
+        if self.vdirsyncer_executable:
+            vdirsyncer_executable = self.vdirsyncer_executable
+        else:
+            vdirsyncer_executable = 'Not found'
+
+        attrs = [
+            ["-- VdirSyncer"],
+            ["Executable: {}".format(vdirsyncer_executable) ],
+            ["Config: {}".format(VDIRSYNCER_CONFIG)]
+        ]
+        Listbox(i=self.i, o=self.o, contents=attrs).activate()
```

Comment:
you need to pass `contents` as the first positional argument. Sorry, I know we need better online docs for that - don't hesitate to refer to the code in `ui/listbox.py` in the meantime =)

Reply:
Also, you want to have a settings menu? I don't quite understand what's happening in this code, I see it's a settings menu, but it doesn't have any way to change anything. It's just for a display, right? If so, want any help with actually allowing changes? It should be easy, `PathPicker` will cover most of your needs here.